integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/management/CarbonIngestCommand.scala

```diff
@@ -0,0 +1,368 @@
+                      s"segment entry and load again")
+          val segmentToWrite = segments.filterNot(_.getLoadName.equals(segmentId))
+          SegmentStatusManager.writeLoadDetailsIntoFile(
+            CarbonTablePath.getTableStatusFilePath(table.getTablePath),
+            segmentToWrite)
+        case other =>
+          throw new RuntimeException(s"Segment $segmentId is in unexpected state: $other")
+      }
+    } finally {
+      if (lock != null) {
+        lock.unlock()
+      }
+    }
+    LOGGER.info(s"Finish recovery, delete snapshot file: $snapshotFilePath")
+    FileFactory.getCarbonFile(snapshotFilePath).delete()
+  }
+
+  /**
+   * Start global sort loading
+   */
+  private def startLoading(
+      spark: SparkSession,
+      table: CarbonTable,
+      loadModel: CarbonLoadModel,
+      stageInput: Seq[StageInput]
+  ): Unit = {
+    val splits = stageInput.flatMap(_.createSplits().asScala)
+    LOGGER.info(s"start to load ${splits.size} files into " +
+                s"${table.getDatabaseName}.${table.getTableName}")
+    val start = System.currentTimeMillis()
+    val dataFrame = DataLoadProcessBuilderOnSpark.createInputDataFrame(spark, table, splits)
+    DataLoadProcessBuilderOnSpark.loadDataUsingGlobalSort(
+      spark,
+      Option(dataFrame),
+      loadModel,
+      SparkSQLUtil.sessionState(spark).newHadoopConf()
+    ).map { row =>
+        (row._1, FailureCauses.NONE == row._2._2.failureCauses)
+      }
+
+    LOGGER.info(s"finish data loading, time taken ${System.currentTimeMillis() - start}ms")
+  }
+
+  /**
+   * Read stage files and return input files
+   */
+  private def collectStageInputs(
+      executorService: ExecutorService,
+      stageFiles: Array[(CarbonFile, CarbonFile)]
+  ): Seq[StageInput] = {
+    val startTime = System.currentTimeMillis()
+    val output = Collections.synchronizedList(new util.ArrayList[StageInput]())
+    val gson = new Gson()
+    stageFiles.map { stage =>
+      executorService.submit(new Runnable {
+        override def run(): Unit = {
+          val filePath = stage._1.getAbsolutePath
+          val stream = FileFactory.getDataInputStream(filePath, FileFactory.getFileType(filePath))
+          val stageInput = gson.fromJson(new InputStreamReader(stream), classOf[StageInput])
+          output.add(stageInput)
+        }
+      })
+    }.map { future =>
+      future.get()
+    }
+    LOGGER.info(s"read stage files taken ${System.currentTimeMillis() - startTime}ms")
+    output.asScala
+  }
+
+  /**
+   * Delete stage file and success file
+   */
+  private def deleteStageFiles(
+      executorService: ExecutorService,
+      stageFiles: Array[(CarbonFile, CarbonFile)]): Unit = {
+    val startTime = System.currentTimeMillis()
+    stageFiles.map { files =>
+      executorService.submit(new Runnable {
+        override def run(): Unit = {
+          files._1.delete()
+          files._2.delete()
```

Comment:
The second file will not be deleted when the first file delete fail, should we consider for the case?

Reply:
In the implementation of delete, it will ignore IOException, so files._2.delete() will be called